ReactSkia/sdk/OnScreenKeyBoard.cpp

```diff
@@ -0,0 +1,646 @@
+        for (unsigned int index=1;index<oskLayout_.keyInfo->at(0).size();index++) {
+            if(oskLayout_.keyInfo->at(0).at(index).KBPartitionId != oskLayout_.keyInfo->at(0).at(index-1).KBPartitionId) {
+                xpos=oskLayout_.keyPos->at(0).at(index).highlightTile.x()-(oskLayout_.keyPos->at(0).at(index).highlightTile.fLeft - oskLayout_.keyPos->at(0).at(index-1).highlightTile.fRight)/2;
+                OSKcanvas_->drawLine(xpos,startY,xpos,endY,paint);
+            }
+        }
+    }
+}
+
+void OnScreenKeyboard ::highlightFocussedKey(SkPoint index) {
+    SkPaint paintObj;
+    unsigned int rowIndex=index.y(),keyIndex=index.x();
+    unsigned int lastRowIndex=lastFocussedKey_.y(),lastKeyIndex=lastFocussedKey_.x();
+
+     RNS_PROFILE_START(HighlightOSKKey)
+     //reset last focussed item
+     paintObj.setColor(bgColor_);
+     paintObj.setAntiAlias(true);
+     OSKcanvas_->drawRect(oskLayout_.keyPos->at(lastRowIndex).at(lastKeyIndex).highlightTile,paintObj);
+     drawFont({lastKeyIndex,lastRowIndex},fontColor_);
+
+    //Hight current focussed item
+    paintObj.setColor(OSK_HIGHLIGHT_BG_COLOR);
+    OSKcanvas_->drawRect(oskLayout_.keyPos->at(rowIndex).at(keyIndex).highlightTile,paintObj);
+    drawFont({keyIndex,rowIndex},OSK_HIGHLIGHT_FONT_COLOR,true);
+    pushToDisplay();
+    RNS_PROFILE_END(" Highlight Completion : ",HighlightOSKKey)
+    return;
+}
+void OnScreenKeyboard::onExposeHandler(RnsShell::Window* window) {
+    if(window  == OSKwindow_) {
+        sem_wait(&semReadyToDraw);
+        if(exposeEventID_) {
+            NotificationCenter::defaultCenter().removeListener(exposeEventID_);
+            exposeEventID_=-1;
+        }
+        showOSKWindow();
+    }
+ }
+
+void OnScreenKeyboard::onHWkeyHandler(rnsKey keyValue, rnsKeyAction eventKeyAction) {
+    if(eventKeyAction != RNS_KEY_Press) return;
+
+    SkPoint hlCandidate;
+    lastFocussedKey_=focussedKey_;
+    unsigned int rowIndex=focussedKey_.y(),keyIndex=focussedKey_.x();
+    RNS_LOG_DEBUG("KEY RECEIVED : "<<RNSKeyMap[keyValue]);
+
+    switch( keyValue ) {
+    /*Case 1: handle Enter/selection key*/
+        case RNS_KEY_Select:
+            handleSelection();
+            lastFocussedKey_=hlCandidate=focussedKey_;
+        break;
+    /* Case  2: Process Navigation Keys*/
+        case RNS_KEY_Right:
+            hlCandidate= oskLayout_.siblingInfo->at(rowIndex).at(keyIndex).siblingRight;
+        break;
+        case RNS_KEY_Left:
+            hlCandidate= oskLayout_.siblingInfo->at(rowIndex).at(keyIndex).siblingLeft;
+        break;
+        case RNS_KEY_Up:
+            hlCandidate= oskLayout_.siblingInfo->at(rowIndex).at(keyIndex).siblingUp;
+        break;
+        case RNS_KEY_Down:
+            hlCandidate= oskLayout_.siblingInfo->at(rowIndex).at(keyIndex).siblingDown;
+        break;
+        default:
+        {
+            bool keyFound=false;
+            hlCandidate=focussedKey_;
+            if(keyValue !=RNS_KEY_UnKnown ) {
```

Comment:
check additional condition to confirm it's a keyboard Key
[>= RNS_KEY_1 && < RNS_KEY_UnKnown]